BallData   ball
PlayerData[] opp_players
PlayerData[] team_players
